beacon_node/http_api/src/publish_blocks.rs

```diff
@@ -238,6 +240,35 @@ pub async fn publish_block<T: BeaconChainTypes, B: IntoGossipVerifiedBlockConten
         }
     }
 
+    if let Some(gossip_verified_data_columns) = gossip_verified_data_columns {
+        let custody_columns = network_globals
+            .custody_columns(block.epoch())
+            .map_err(|e| {
+                warp_utils::reject::broadcast_without_import(format!(
+                    "Failed to compute custody column indices: {:?}",
+                    e
+                ))
+            })?;
+
+        for data_column in gossip_verified_data_columns {
+            if custody_columns.contains(&data_column.index()) {
+                if let Err(e) = Box::pin(chain.process_gossip_data_column(data_column)).await {
+                    let msg = format!("Invalid data column: {e}");
+                    return if let BroadcastValidation::Gossip = validation_level {
+                        Err(warp_utils::reject::broadcast_without_import(msg))
+                    } else {
+                        error!(
+                            log,
+                            "Invalid blob provided to HTTP API";
+                            "reason" => &msg
+                        );
+                        Err(warp_utils::reject::custom_bad_request(msg))
+                    };
+                }
+            }
+        }
+    }
```

Comment:
Should you import all the columns when producing a block? I see that there's no mechanism for others to retrieve data you created but don't declare to custody. However it frightens me a little that network liveness depends on those very few peers of the block proposer that are meant to custody specific rows. If a proposer happens to not be peered at the time of publication with a peer on each subnet there will be data loss.

Reply:
Added this to tracking issue https://github.com/sigp/lighthouse/issues/4983